### DISCUSSION TITLE
Boundary Conditions not getting implemented

### DISCUSSION THREAD

**Discussion by gitvicky** [zwicker-group/py-pde#58] [Q&A]
Hi, 

I was testing out the API and noticed that the boundary conditions are not being implemented. With or without the mentioned BC the solutions are the same. 

```
# 2D Diffusion test 
D = 0.01
eq = PDE(
    {
        "u": f"{D}*laplace(u)"
    }
)


grid = UnitGrid([4, 4])
field = ScalarField.random_normal(grid, mean=5, seed=42)
bc = {'type': 'value', 'value': 3}
field.laplace(bc)



tracker = PlotTracker(interval=1)
storage = MemoryStorage()
sol = eq.solve(field, t_range=20, dt=1e-3, tracker=[tracker, storage.tracker(1)])
```

**Comment by david-zwicker** [comment]
I'm not quite sure what you're trying to accomplish, but if you want to apply the boundary conditions to the PDE, you need to specify it in the [initialization of the `PDE` class using the `bc` argument as described in the documentation](https://py-pde.readthedocs.io/en/latest/packages/pde.pdes.html#pde.pdes.pde.PDE).

**Comment by gitvicky** [comment]
Hey, 
 I was adding on the Boundary Conditions as I had seen [here](https://py-pde.readthedocs.io/en/latest/manual/advanced_usage.html#boundary-conditions)

So for all cases, there is only need to specify the boundary condition in a list and add them to the PDE construct ? What does this field.laplace do then ? 

Thanks so much for the prompt responses.

**Reply by david-zwicker** [reply]
`field.laplace(bc)` applies the Laplace operator to the `field`; see the documentation. This call returns another `ScalarField`.

**Comment by gitvicky** [comment]
Understood. Thank you very much.
